modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb

```diff
@@ -391,23 +400,53 @@ def international_phone_too_long?
         phone.length > 25 if phone
       end
 
-      def validate_form_526_gulf_service
+      def validate_form_526_toxic_exposure
+        return if form_attributes&.dig('toxicExposure').nil? # nullable on schema
+
         gulf_war_service = form_attributes&.dig('toxicExposure', 'gulfWarHazardService')
-        return if gulf_war_service&.dig('servedInGulfWarHazardLocations') == 'NO'
+        validate_form_526_toxic_exp_sections(gulf_war_service, 'gulfWarHazardService')
+        herbicide_service = form_attributes&.dig('toxicExposure', 'herbicideHazardService')
+        validate_form_526_toxic_exp_sections(herbicide_service, 'herbicideHazardService')
+        other_exposures = form_attributes&.dig('toxicExposure', 'additionalHazardExposures')
+        validate_form_526_toxic_exp_sections(other_exposures, 'additionalHazardExposures')
+        multi_exposures = form_attributes&.dig('toxicExposure', 'multipleExposures')
+        validate_form_526_toxic_multi_exp(multi_exposures, 'multipleExposures')
+      end
+
+      def validate_form_526_toxic_exp_sections(section, attribute_name)
+        if section&.nil? || section&.dig('servedInHerbicideHazardLocations') == 'NO' ||
+           section&.dig('servedInGulfWarHazardLocations') == 'NO'
+          return
+        end
+
+        begin_date = section&.dig('serviceDates', 'beginDate')
+        end_date = section&.dig('serviceDates', 'endDate')
+
+        begin_prop = "/toxicExposure/#{attribute_name}/serviceDates/beginDate"
+        end_prop = "/toxicExposure/#{attribute_name}/serviceDates/endDate"
 
-        begin_date = gulf_war_service&.dig('serviceDates', 'beginDate')
-        end_date = gulf_war_service&.dig('serviceDates', 'endDate')
+        validate_service_date(begin_date, begin_prop) unless begin_date.nil? || !date_is_valid?(begin_date,
+                                                                                                begin_prop)
+        validate_service_date(end_date, end_prop) unless end_date.nil? || !date_is_valid?(end_date, end_prop)
+      end
+
+      def validate_form_526_toxic_multi_exp(section, attribute_name)
+        return if section.nil?
+
+        section&.each do |item, idx|
+          begin_date = item&.dig('serviceDates', 'beginDate')
+          end_date = item&.dig('serviceDates', 'endDate')
```

Comment:
Not sure this will work. Dates section of `multipleExposures` is defined as `exposureDates`, not `serviceDates`.